core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/principal/Principal.scala

```diff
@@ -98,6 +100,8 @@ class Principal(val metadata: OpExecConfig)
   implicit val ec: ExecutionContext = context.dispatcher
   implicit val timeout: Timeout = 5.seconds
 
+  override lazy val promiseManager: PromiseManager = wire[PromiseManager]
```

Comment:
In both controller and principal, we have put promiseManagers. Let's remove it. Only the Data/ControlInputPorts and I would assume the NetworkSenderActors should have promisemanagers.

Reply:
all the promise handlers are in promise managers right now, if you move it to NetworkSenderActors, how will it interact with the dp thread and all other things?

Reply:
PromiseManager should go to the ports. Only handlers should be here